src/main/java/org/samoxive/safetyjim/discord/DiscordShard.kt

```diff
@@ -347,6 +348,49 @@ class DiscordShard(private val bot: DiscordBot, shardId: Int, sessionController:
             }
         }
 
+        if (guildSettings.invitelinkremover){
+            val blacklistedHosts = arrayOf("discord.gg/")
+            for (blacklistedHost in blacklistedHosts) {
+                if (member.contains(blacklistedHost)) {
+
+                    val kickUser = member;
+
+                    val embed = EmbedBuilder()
+                    embed.setTitle("Kicked from " + guild.name)
+                    embed.setColor(Color(0x4286F4))
+                    embed.setDescription("You were kicked from " + guild.name)
+                    embed.addField("Reason: For username that contains an invite link" , false)
+                    embed.setFooter("Kicked by " + bot, null)
+                    embed.setTimestamp(now.toInstant())
+
+                    kickUser.sendDM(embed.build())
+
+                    try {
+                        val auditLogReason = String.format("Kicked by %s - %s", user.getUserTagAndId(), reason)
```

Comment:
`user` is the user that just joined the server, the reason text makes it look like they kicked themselves. You can get Jim's member object in the guild with `guild.selfMember`, which should contain the user object as well.